### PR TITLE
CORDA-579: Add EdDSA engine that understands X.509 keys

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

**Comment by rnicoll** [comment]
Works around an issue with deserialized X.509 certificates containing EdDSA keys that the library we use cannot accept. A permanent fix should go into the library, this is just an interim solution.

**Comment by mnesbit** [comment]
This does look essential for 1.0. Essentially any certificates loaded from the keystore that contain an EdDSA key will fail to decode. Thus after a node reboot no further generated anonymous keys will validate.

File: core/src/main/kotlin/net/i2p/crypto/eddsa/KludgeEdDSAEngine.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package net.i2p.crypto.eddsa
+
+import java.security.*
+import java.security.spec.AlgorithmParameterSpec
+import java.security.spec.X509EncodedKeySpec
+
+/**
+ * Temporary solution to EdDSA engine not being able to handle X.509 keys.
+ */
+class KludgeEdDSAEngine : Signature {
```
Comment (mnesbit) [inline core/src/main/kotlin/net/i2p/crypto/eddsa/KludgeEdDSAEngine.kt]: Kludge is not a good word. Perhaps X509EdDSAEngine?

**Comment by rnicoll** [comment]
Rename class, ready for re-review

**Comment by mnesbit** [comment]
We should raise an issue on i2p and include the tracking link in the comments

**Comment by rnicoll** [comment]
Raised str4d/ed25519-java#47 upstream as hopefully a permanent fix.

**Pull request merged**